### PR TITLE
fix writing issue

### PR DESCRIPTION
**Pull request by gfcapalbo**
every thing worked fine for external collaborators. but when timesheets are active , writing on task will activate timesheet. Timesheets need a user. overwriting create to manage that too.

### CODE REVIEW

File: trp_project_scrum/models/project_task_work.py [hunk 1]
```diff
@@ -41,3 +41,14 @@ def get_user_related_details(self, user_id):
 
         res = super(ProjectTaskWork, self).get_user_related_details(user_id)
         return res
+
+        @api.model
+        def create(self, vals):
+            user=self.env['res.users'].browse(self._uid)
```
Comment (NL66278) [inline trp_project_scrum/models/project_task_work.py]: No need for this as you can use self.env.user.

File: trp_project_scrum/models/project_task_work.py [hunk 2]
```diff
@@ -41,3 +41,14 @@ def get_user_related_details(self, user_id):
...
+            if not user.sudo().employee_ids:
```
Comment (NL66278) [inline trp_project_scrum/models/project_task_work.py]: Are you certain you need sudo here?

**Comment by NL66278** [comment]
Did you test this?

File: trp_project_scrum/models/project_task_work.py (hunk 2 above)
Comment (gfcapalbo) [reply trp_project_scrum/models/project_task_work.py]: yes we did not give permissions for anything other than the minimum so "collaborator" has no access to hr.employee .  you can see it is used elswhere to read the information. it is not writing just reading.

File: trp_project_scrum/models/project_task_work.py (hunk 1 above)
Comment (gfcapalbo) [reply trp_project_scrum/models/project_task_work.py]: yes self._uid   is a shortcut nobody likes , i will change it.

**Comment by gfcapalbo** [comment]
yes it is deployed on help test and before installing it I refreshed the DB with the latest production DB ( cronjobs disabled)

**Comment by gfcapalbo** [comment]
You can see it in action on helptest by entering as pedro or test_collaborator ort jordi,.

**Comment by gfcapalbo** [comment]
@NL66278 simplified user fetching as suggested.

File: trp_project_scrum/security/ir_rules_collaborators_followers.xml [hunk 3]
```diff
@@ -97,4 +97,17 @@
             <field name="perm_create" eval="False" />
         </record>
     </data>
+
+        <record id="test_hr_employee_foll" model="ir.rule">
+            <field name="name">My employee read (if any)</field>
+            <field name="model_id" ref="hr.model_hr_employee"/>
+            <field name="domain_force">[
+                ('address_id', 'in', [user.partner_id.id] )]</field>
```
Comment (hbrunn) [inline trp_project_scrum/security/ir_rules_collaborators_followers.xml]: `('user_id', '=', user.id)`

Comment (gfcapalbo) [reply trp_project_scrum/security/ir_rules_collaborators_followers.xml]: 802ec772b7e67dc102e066a8dda201f639d36688

**Review by gfcapalbo — COMMENTED** [review]
@hbrunn  done. extensivley tested on a mirrored production DB on local.

**Comment by gfcapalbo** [comment]
@hbrunn passing rules added for all models involved , productproduct , producttemplate productuom and pricetype. strictly read and only for current product on time sheet . tested on local .

**Comment by hbrunn** [comment]
superseded by https://github.com/Therp/Therp-Addons/pull/66

**Pull request closed without merging**